cosipy/response/FullDetectorResponse.py

```diff
@@ -39,63 +31,423 @@ class FullDetectorResponse(HealpixBase):
+        with gzip.open(filename, "rt") as file:
+             
+            progress_bar = tqdm(file, total=nlines, desc="Progress", unit="line")
+
+            for line in progress_bar:
+
+                
+                line = line.split()
+
+                if len(line) == 0:
+                    continue
+
+                key = line[0]
+
+                if key == 'RD':
+
+                    b = np.array(line[1:-1], dtype=int)
+                    c = int(line[-1])
+
+                    coords[:, sbin] = b
+                    data[sbin] = c
+
+                    sbin += 1
+                    
+                progress_bar.update(1)
+        
+        progress_bar.close()
+        nbins_sparse = sbin
+
+        print("response file read ! Now we create the histogram and weight in order to "+ 
+                "get the effective area")
+        # create histpy histogram
+        dr = Histogram(axes, contents=COO(coords=coords[:, :nbins_sparse], data= data[:nbins_sparse], shape = tuple(axes.nbins)))
+
+        
+
+        # Weight to get effective area
+
+        ewidth = dr.axes['Ei'].widths
+        ecenters = dr.axes['Ei'].centers
+
+        if Spectrumfile is not None and norm=="file":
+            print("normalisation : spectrum file")
+            # From spectrum file
+            spec = pd.read_csv(Spectrumfile, sep=" ")
+            spec = spec.iloc[:-1]
+            hspec = Histogram([h_spec.axes[1]])
+            hspec[:] = np.interp(hspec.axis.centers,
+                             spec.iloc[:, 0].to_numpy(),
+                             spec.iloc[:, 1].to_numpy(),
+                             left=0,
+                             right=0) * ewidth
+            hspec /= np.sum(hspec)
+
+            nperchannel_norm = hspec[:]
+
+        elif norm=="powerlaw":
+            print("normalisation : powerlaw with index {0} with energy range [{1}-{2}]keV".format(alpha,emin,emax))
+            # From powerlaw
+            
+
+            if alpha == 1:
+                K = 1 / np.log(emax/emin)
+            else:
+                K = (1-alpha) / (emax**(1-alpha) - emin**(1-alpha))
+
+            nperchannel_norm = K * ecenters**(-alpha) * ewidth
+
+            nperchannel_norm[ecenters < emin] = 0
+            nperchannel_norm[ecenters > emax] = 0
+
+
+        elif norm =="Linear" :
+            print("normalisation : linear with energy range [{0}-{1}]".format(emin,emax))
+            nperchannel_norm = ewidth / (emax-emin)
+            
+        elif norm=="Mono" :
+            print("normalisation : mono")
+            nperchannel_norm = 1.
+            
+        nperchannel = nperchannel_norm * nevents_sim
+
+        # Full-sky?
+        if not single_pixel:
+
+            # Assumming all FISBEL pixels have the same area
+            nperchannel /= dr.axes['NuLambda'].nbins
+
+        # Area
+        counts2area = area_sim / nperchannel
+        dr_area = dr * dr.expand_dims(counts2area, 'Ei')
```

Comment:
When I run this with mono normalization, I get an error saying that the dimensions of dr['Ei'] and counts2area don't match up. I think this is because counts2area is a float.